interfaces/builtin/compatibility.go

```diff
@@ -45,9 +45,18 @@ type CompatField struct {
 	Dimensions []CompatDimension
 }
 
+// CompatSpec specifies valid values for a compatibility field, this can be
```

Comment:
This part is entirely new to me. Would you mind explaining this next week when you are back (perhaps interactively so we can use a high-bandwidth channel).